modules/dxp/apps/saml/saml-impl/src/main/java/com/liferay/saml/runtime/internal/messaging/SamlMessageListener.java

```diff
@@ -14,14 +14,15 @@
 
 package com.liferay.saml.runtime.internal.messaging;
 
-import com.liferay.portal.kernel.messaging.BaseMessageListener;
+import com.liferay.portal.kernel.messaging.BaseMessageListenerCompanyScope;
 import com.liferay.portal.kernel.messaging.Message;
 import com.liferay.portal.kernel.messaging.MessageListenerException;
 
 /**
  * @author Tomas Polesovsky
  */
-public abstract class SamlMessageListener extends BaseMessageListener {
+public abstract class SamlMessageListener
```

Comment:
@achaparro  
Why this is a company scope listener? The impl is wrong, it is a loop.

SamlMessageListener#doReceive(message, long) -> BaseMessageListenerCompanyScope#doReceive(message) ->
SamlMessageListener#doReceive(message, long)

Reply:
@tinatian,
When we call this:
SamlMessageListener#doReceive(message, long) we will call the child implementation of doReceive(message) so it won't be an infinite loop

For SamlMetadataMessageListener we can keep the loop the by company when db partition is not enabled and for the other samllisteners, they will be executed once when db partition is not enabled and as many times as companies when db partition is enabled.